src/main/java/io/supertokens/Main.java

```diff
@@ -209,16 +209,14 @@ private void init() throws IOException, StorageQueryException {
             // load all configs for each of the tenants.
             MultitenancyHelper.getInstance(this).loadConfig(new ArrayList<>());
 
-            // We now want all tenant entries to be synced on their appropriate storages
-            TenantConfig[] tenantConfigs = MultitenancyHelper.getInstance(
-                    this).getAllTenants();
-            List<TenantIdentifier> tenantIdentifiers = new ArrayList<>();
-            for (TenantConfig tenantConfig : tenantConfigs) {
-                tenantIdentifiers.add(tenantConfig.tenantIdentifier);
-            }
+            // we want to load storage layer once again so that the base storage also contains the right
+            // tenant identifier set passed to the init. So we close the base storage layer and also clear
+            // all the resources for storage layer
+            StorageLayer.getBaseStorage(this).stopLogging();
+            StorageLayer.getBaseStorage(this).close();
```

Comment:
what will happen if this fails? For example, if the initial initStorage failed, then will this fail too?

Reply:
if initStorage fails on the base tenant, the core will crash. we can proceed further only if the base tenant storage can load successfully.